packages/webapp/src/containers/ErrorHandler/saga.js

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright (C) 2007 Free Software Foundation, Inc. <https://fsf.org/>
+ *  This file (saga.js) is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { createAction } from '@reduxjs/toolkit';
+import { call, select, takeLeading } from 'redux-saga/effects';
+import { userFarmSelector } from '../userFarmSlice';
+import { getUserFarmsSaga } from '../ChooseFarm/saga';
+import { logout } from '../../util/jwt';
+
+
+export const handle403 = createAction(`handle403Saga`);
+
+export function* handle403Saga() {
+  yield call(getUserFarmsSaga);
+  const { has_consent } = yield select(userFarmSelector);
+  if (has_consent) {
```

Comment:
I don't understand all the details, but it seems strange to say "if we have the users consent (to the terms of use) then log them out". Maybe I'm interpreting that wrong, or it makes sense in this particular context.

Reply:
it first sync store with db then
It checks whether 403 is from has_consent === false. 
If status === 'inactive' or store info is incorrect fetch requests also return 403, then user will be logged out.

Reply:
Should be fine